### ISSUE TITLE
dot disconnection combined with branches and closures

### ISSUE THREAD

**Issue by adalke** [baoilleach/deepsmiles#6]
I think that the dot disconnection "." should reset the system state so that closures and branches cannot cross it.

Here's an example of using a branch to cross a dot:
```
>>> import deepsmiles
>>> conv = deepsmiles.Converter(True, True)
>>> conv.decode("CN.OP))S")
'CN.(OP)S'
>>> conv.decode("CN.OP)))S")
'CN(.OP)S'
```
In both cases, the result is not a valid SMILES.

Here is an example of using a closure to cross a dot:
```
>>> import deepsmiles
>>> conv = deepsmiles.Converter(True, True)
>>> conv.decode("C.C2")
'C.1C1'
```
You'll note that the result here is also not a valid SMILES.

While it is possible to fix the code to support these use cases, I think it's better to disallow dot disconnections to be used this way.

We both know that dot disconnections like this are useful in SMILES. However, the uses I can think of - like simple combinatorial generation - only work with closures, and depend on being able to label both sides of disconnection. That's not possible with DeepSMILES.

**Comment by baoilleach** [comment]
I agree (after some thought). The deepsmiles syntax doesn't allow arbitary linkages between atoms even in the normal case - it only allows linkages to atoms that preceded it in the DFS tree. So linking to atoms in a completely different tree should be right out.

**Comment by adalke** [comment]
Yes, I think the keep simplification of DeepSMILES is its close adherence to a DFS tree, while SMILES is a more free-form grammar.
